tools/rosbag/src/rosbag/bag.py

```diff
@@ -1580,6 +1580,7 @@ def _read_uint32(f): return _unpack_uint32(f.read(4))
 def _read_uint64(f): return _unpack_uint64(f.read(8))
 def _read_time  (f): return _unpack_time  (f.read(8))
 
+def _unpack_str(v):    return v.decode()
```

Comment:
The function name doesn't make much sense in the case of Python 3. Since you invoke `decode` on `v` it can't be a `str` but must be `bytes`?